Bottom: 170820b8c95d6b36a07975fb846ba32cf6d1ca3c
Top:    cb36d77503832f2e17c97978e8ccfb7d0594aecf
Author: Jason Wang <jasowang@redhat.com>
Date:   2015-09-28 12:31:19 +0800

e1000: use alias for default model

Instead of using a new type for default model, using an alias for this
to avoid bit duplication.

Cc: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/hw/alpha/dp264.c b/hw/alpha/dp264.c
index 27bdaa1..e993be6 100644
--- a/hw/alpha/dp264.c
+++ b/hw/alpha/dp264.c
@@ -87,7 +87,7 @@ static void clipper_init(MachineState *machine)
 
     /* Network setup.  e1000 is good enough, failing Tulip support.  */
     for (i = 0; i < nb_nics; i++) {
-        pci_nic_init_nofail(&nd_table[i], pci_bus, "e1000", NULL);
+        pci_nic_init_nofail(&nd_table[i], pci_bus, "e1000-82540em", NULL);
     }
 
     /* IDE disk setup.  */
diff --git a/hw/i386/pc.c b/hw/i386/pc.c
index 461c128..ce68cba 100644
--- a/hw/i386/pc.c
+++ b/hw/i386/pc.c
@@ -1583,7 +1583,7 @@ void pc_nic_init(ISABus *isa_bus, PCIBus *pci_bus)
         if (!pci_bus || (nd->model && strcmp(nd->model, "ne2k_isa") == 0)) {
             pc_init_ne2k_isa(isa_bus, nd);
         } else {
-            pci_nic_init_nofail(nd, pci_bus, "e1000", NULL);
+            pci_nic_init_nofail(nd, pci_bus, "e1000-82540em", NULL);
         }
     }
 }
diff --git a/hw/net/e1000.c b/hw/net/e1000.c
index 09c9e9d..7e87249 100644
--- a/hw/net/e1000.c
+++ b/hw/net/e1000.c
@@ -1666,11 +1666,6 @@ static const E1000Info e1000_devices[] = {
     },
 };
 
-static const TypeInfo e1000_default_info = {
-    .name          = "e1000",
-    .parent        = "e1000-82540em",
-};
-
 static void e1000_register_types(void)
 {
     int i;
@@ -1688,7 +1683,6 @@ static void e1000_register_types(void)
 
         type_register(&type_info);
     }
-    type_register_static(&e1000_default_info);
 }
 
 type_init(e1000_register_types)
diff --git a/hw/pci/pci.c b/hw/pci/pci.c
index b0bf540..1762b06 100644
--- a/hw/pci/pci.c
+++ b/hw/pci/pci.c
@@ -1637,7 +1637,7 @@ static const char * const pci_nic_models[] = {
     "i82557b",
     "i82559er",
     "rtl8139",
-    "e1000",
+    "e1000-82540em",
     "pcnet",
     "virtio",
     NULL
@@ -1649,7 +1649,7 @@ static const char * const pci_nic_names[] = {
     "i82557b",
     "i82559er",
     "rtl8139",
-    "e1000",
+    "e1000-82540em",
     "pcnet",
     "virtio-net-pci",
     NULL
diff --git a/hw/ppc/ppc440_bamboo.c b/hw/ppc/ppc440_bamboo.c
index ada676c..f45fc1c 100644
--- a/hw/ppc/ppc440_bamboo.c
+++ b/hw/ppc/ppc440_bamboo.c
@@ -246,7 +246,7 @@ static void bamboo_init(MachineState *machine)
         for (i = 0; i < nb_nics; i++) {
             /* There are no PCI NICs on the Bamboo board, but there are
              * PCI slots, so we can pick whatever default model we want. */
-            pci_nic_init_nofail(&nd_table[i], pcibus, "e1000", NULL);
+            pci_nic_init_nofail(&nd_table[i], pcibus, "e1000-82540em", NULL);
         }
     }
 
diff --git a/qdev-monitor.c b/qdev-monitor.c
index 0bf7f83..69bce78 100644
--- a/qdev-monitor.c
+++ b/qdev-monitor.c
@@ -50,6 +50,7 @@ static const QDevAlias qdev_alias_table[] = {
     { "lsi53c895a", "lsi" },
     { "ich9-ahci", "ahci" },
     { "kvm-pci-assign", "pci-assign" },
+    { "e1000-82540em", "e1000"},
     { }
 };
